### PR TITLE
Adding bundling support for PublisherApi.publish

### PR DESCRIPTION
**Pull request by garrettjonesgoogle**

### CODE REVIEW

File: gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherSettings.java [hunk 1]
```diff
@@ -223,7 +232,7 @@ protected PublisherSettings(MethodBuilders methods) {
   }
 
   /**
-   * Returns the ApiCallableBuilder for the API method createTopic.
+   * Returns the builder for the API method createTopic.
```
Comment (aozarov) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherSettings.java]: Did I mention that I think you should consider renaming `ApiCallableBuilder` to `Builder`? As a nested class I think
its context is clear and such naming convention is very common ([e.g.](https://github.com/google/guava/blob/master/guava/src/com/google/common/collect/ImmutableList.java#L612)).

File: gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/SubscriberApi.java [hunk 2]
```diff
@@ -66,9 +65,26 @@
  */
 @javax.annotation.Generated("by GAPIC")
 public class SubscriberApi implements AutoCloseable {
+  // ========
```
Comment (aozarov) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/SubscriberApi.java]: Per earlier discussion (I don't remember the status). Are these comments really necessary for the code-generation part? (This is not a common practice).

File: gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherSettings.java (hunk 1 above)
Comment (garrettjonesgoogle) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherSettings.java]: I don't think you mentioned that yet. I'll go ahead and make that change in GAX. Would you like to see that in this PR, or is a following PR ok?

Comment (aozarov) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherSettings.java]: following PR sounds good to me.

File: gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherSettings.java [hunk 3]
```diff
@@ -337,4 +346,66 @@ public Object extractNextToken(ListTopicSubscriptionsResponse payload) {
...
+
+        @Override
+        public void splitResponse(
+            PublishResponse bundleResponse,
+            Collection<? extends RequestIssuer<PublishRequest, PublishResponse>> bundle) {
+          int bundleMessageIndex = 0;
+          for (RequestIssuer<PublishRequest, PublishResponse> responder : bundle) {
+            List<String> subresponseElements = new ArrayList<>();
+            int subresponseCount = responder.getRequest().getMessagesCount();
+            for (int i = 0; i < subresponseCount; i++) {
```
Comment (aozarov) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherSettings.java]: Any reason for not using for each loop? Is response.getRequest().getMessageCount() can be less
than the size of bundleResponse.getMessageIdsList()?

Also (nit pick), you can Initialize subresponseElements to the expected size if you already know it.

**Comment by aozarov** [comment]
Looks good. Feel free to merge unless you want to apply any of the suggested changes in this PR.

File: gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherSettings.java (hunk 3 above)
Comment (garrettjonesgoogle) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherSettings.java]: Any time more than 1 PublishRequest is bundled, response.getRequest().getMessageCount() will be less than bundleResponse.getMessageIdsList(). So if PublishRequest A has M messages and PublishRequest B has N messages, then response.getRequest().getMessageCount() will return M and then N.

File: gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/SubscriberApi.java (hunk 2 above)
Comment (garrettjonesgoogle) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/SubscriberApi.java]: I don't recall an objection to them before. They make it less likely for there to be change conflicts that go across multiple sections.

Comment (aozarov) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/SubscriberApi.java]: Less likely by the code-generator? Again, feel free to look at other common java libraries (guava, and JDK included).
This is by far not a common practice.

Comment (garrettjonesgoogle) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/SubscriberApi.java]: Less likely by the 3-way merge process.

They are really just a holdover from the coding style of the first prototype; I'll see what the team thinks and possibly remove them in the next version (which will have other style fixes).

**Pull request merged**